Reference/google/protobuf/any.pb.swift

```diff
@@ -108,18 +108,14 @@ import Foundation
 ///         "@type": "type.googleapis.com/google.protobuf.Duration",
 ///         "value": "1.212s"
 ///       }
-struct Google_Protobuf_Any: ProtobufGeneratedMessage, ProtobufProto3Message {
+struct Google_Protobuf_Any: ProtobufGeneratedMessage, ProtobufProto3Message, ProtoNameProviding {
   public var swiftClassName: String {return "Google_Protobuf_Any"}
   public var protoMessageName: String {return "Any"}
   public var protoPackageName: String {return "google.protobuf"}
-  public var jsonFieldNames: [String: Int] {return [
-    "typeUrl": 1,
-    "value": 2,
-  ]}
-  public var protoFieldNames: [String: Int] {return [
-    "type_url": 1,
-    "value": 2,
-  ]}
+  public static let _protobuf_fieldNames: FieldNameMap = [
+    1: .unique(proto: "type_url", json: "typeUrl", swift: "typeURL"),
+    2: .same(proto: "value", swift: "value"),
```

Comment:
Mirrors are used by things other than debugging, of course, though I hope that people won't use our mirror to tie our objects into some other JSON serialization library.  😱

As for #149, I'm a little torn to be honest.  I agree that slimming things down is good, but I also place a high value on features (in case you hadn't noticed).  And the approach you've taken here suggests that simply deduplicating the JSON and Swift names could eliminate a lot -- maybe even most -- of the remaining overhead.